### PR TITLE
fix: move to query v4

### LINKED ISSUE #190: Move to Query v4
Upgrading @tanstack/query to v4 will allow us to use react 18.
But we should be careful! Uppy is not compatible with React18 (to be checked).

### PR DESCRIPTION
**Pull request by spaenleh**
fix #190

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -8,10 +8,11 @@
...
   "exports": {
-    "types": "./dist/index.d.ts",
-    "import": "./dist/index.js",
-    "require": null,
-    "default": "./dist/index.js"
+    ".": {
+      "types": "./dist/index.d.ts",
+      "import": "./dist/index.js",
+      "default": "./dist/index.js"
+    }
```
Comment (pyphilia) [inline package.json:15]: Why the change?

**Review by pyphilia — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (spaenleh) [reply package.json:15]: I am not sure, but in this version it is using subpath exports: https://nodejs.org/api/packages.html#subpath-exports in case we want to export other entry points later ?
